### DISCUSSION TITLE
Differentiate which event invoked the activate function

### DISCUSSION THREAD

**Discussion by alagiribuddhar** [microsoft/vscode-discussions#349] [Extension Development QnA] [answered]
I am writing a extension with some handlers in activate function. I need add specific handler based on activation event types. So fay only Extension Context is provided. Is there a way to differentiate the activation events which invoked activate function.

**Comment by r3m0t** [comment]
Can you explain why you need that? The activation function is only called once so you need to register every command immediately in case the user invokes additional commands. This applies to most other things you would do at activation.

**Reply by alagiribuddhar** [reply]
Wanted to show user information when particular file is present in workspace. So added onContainsFile activation event and it needs to be shown only when particular file is present and wanted to differentiate from onStart or onStop activation events.

**Reply by isidorn** [reply, marked as answer]
Not possible currently.

**Reply by JustinGrote** [reply]
@isidorn, in my case, because of https://github.com/microsoft/vscode/issues/28122, I need a way to start my extension and find tests so I now need to add `onStartupFinished` as a lazy extension load and start minimal filewatchers there. It would be helpful to know if my extension was started by `onStartupFinished` vs. a file activation event, it would allow me to be more efficient in my loading.
